assets/src/settings-page/supported-templates.js

```diff
@@ -28,15 +43,37 @@ function PostTypeCheckbox( { postTypeObject } ) {
 
 	const {
 		supported_post_types: supportedPostTypes,
+		supportable_templates: supportableTemplates,
+		supported_templates: supportedTemplates,
+		all_templates_supported: allTemplatesSupported,
 	} = editedOptions || {};
 
+	const hasPageOnFront = hasFrontPageTemplate( supportableTemplates );
+	const isBlogTemplateSupported = supportedTemplates.includes( 'is_home' );
+	const isFrontPageTemplateSupported = supportedTemplates.includes( 'is_front_page' );
+
 	return (
 		<li key={ `supportable-post-type-${ postTypeObject.name }` }>
 			<CheckboxControl
 				checked={ supportedPostTypes.includes( postTypeObject.name ) }
 				label={ postTypeObject.label }
 				onChange={
 					( newChecked ) => {
+						if ( ! newChecked && hasPageOnFront && ! allTemplatesSupported && 'page' === postTypeObject.name ) {
+							let warning = '';
+							if ( isBlogTemplateSupported && isFrontPageTemplateSupported ) {
+								warning = __( 'Note that disabling Pages will prevent you from serving your homepage and posts page (blog index) as AMP.', 'amp' );
```

Comment:
Common or capital `P`?

```suggestion
								warning = __( 'Note that disabling pages will prevent you from serving your homepage and posts page (blog index) as AMP.', 'amp' );
```